src/msg/async/ProtocolV2.cc

```diff
@@ -2261,6 +2261,8 @@ CtPtr ProtocolV2::_auth_bad_method(int r)
   std::vector<uint32_t> allowed_modes;
   messenger->auth_server->get_supported_auth_methods(
     connection->get_peer_type(), &allowed_methods, &allowed_modes);
```

Comment:
Continuing to pass `allowed_modes` here is pointless and can actually be confusing, leading to questions like "what is the difference between the modes that are returned by `get_supported_auth_methods()` and the modes that are returned by `get_supported_con_modes()`", "does `allowed_modes` get appended to in `get_supported_con_modes()`", etc.
```suggestion
    connection->get_peer_type(), &allowed_methods);
```

Reply:
I'd go as far as dropping `modes` parameter from `get_supported_auth_methods()` signature so that it's clear that this function only deals with methods/protocols and that connection modes are supposed to be obtained with `get_supported_con_modes()`.

Reply:
:+1: to the Ilya's request.

Perhaps dropping `modes` from `get_supported_auth_methods()` would allow to refactor `AuthRegistry::get_supported_method()` and drop its `switch` nesting as well.

Otherwise LGTM.